react-app/src/components/property/PropertyForms.tsx

```diff
@@ -491,50 +548,105 @@ interface IAssessedValue {
+    // Collect all the year values from the form
+    const yearValues = currentValues.Evaluations.map((evaluation) => parseInt(evaluation.Year));
+
+    // Check for duplicates in the yearValues array
+    const yearCounts = yearValues.reduce((acc, year) => {
+      acc[year] = (acc[year] || 0) + 1;
+      return acc;
+    }, {});
+
+    for (const year in yearCounts) {
+      if (yearCounts[year] > 1) {
+        return `There is already an assessment value for the year ${year}`;
+      }
+    }
+    return true;
+  };
+
   return (
     <Box display={'flex'} flexDirection={'column'} gap={'1rem'}>
       <Typography mt={2} variant="h5">
         {title ?? 'Assessed Value'}
       </Typography>
-      {!years.length ? (
-        <Typography alignSelf={'center'}>No values.</Typography>
-      ) : (
-        <Box overflow={'auto'} paddingTop={'8px'}>
-          {years.map((yr, idx) => {
-            return (
-              <Box
-                mb={2}
-                gap={2}
-                key={`assessedvaluerow-${yr}${'-' + topLevelKey}`}
-                display={'flex'}
-                width={'100%'}
-                flexDirection={'row'}
-              >
-                <TextFormField
-                  sx={{ minWidth: 'calc(33.3% - 1rem)' }}
-                  name={`${topLevelKey ?? ''}Evaluations.${idx}.Year`}
-                  label={'Year'}
-                  value={yr}
-                  disabled
-                />
-                <TextFormField
-                  InputProps={{
-                    startAdornment: <InputAdornment position="start">$</InputAdornment>,
-                  }}
-                  sx={{ minWidth: 'calc(33.3% - 1rem)' }}
-                  name={`${topLevelKey ?? ''}Evaluations.${idx}.Value`}
-                  numeric
-                  label={'Value'}
-                />
-              </Box>
-            );
-          })}
+      <Box overflow={'auto'} paddingTop={'8px'}>
+        {/* Render the current year row first */}
+        <Box
+          mb={2}
+          gap={2}
+          display={'flex'}
+          width={'100%'}
+          flexDirection={'row'}
+          key={`assessedvaluerow-current${'-' + topLevelKey}`}
+        >
+          <TextFormField
+            sx={{ minWidth: 'calc(33.3% - 1rem)' }}
+            name={`${topLevelKey ?? ''}Evaluations.${hasCurrentYear ? 0 : currentYear}.Year`}
+            label={'Year'}
+            value={currentYear}
+            onBlur={handleYearChange}
+            rules={{
+              validate: () => {
+                const result = handleYearChange();
+                return result;
+              },
+            }}
+          />
+          <TextFormField
+            InputProps={{
+              startAdornment: <InputAdornment position="start">$</InputAdornment>,
+            }}
+            sx={{ minWidth: 'calc(33.3% - 1rem)' }}
+            name={`${topLevelKey ?? ''}Evaluations.${hasCurrentYear ? 0 : currentYear}.Value`}
```

Comment:
For these names, are we only ever showing the most recent two? If so, we could probably just use 0 and 1 as the indexes.